client/verta/verta/pipeline/_registered_pipeline.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+
+import json
+import copy
+from typing import Any, Dict, Optional
+
+from verta._internal_utils._utils import Configuration, Connection
+from verta.endpoint.resources import Resources
+from verta.pipeline import PipelineGraph
+from verta.registry.entities import RegisteredModelVersion
+
+
+class RegisteredPipeline:
+    """
+    Object representing a version of a registered inference pipeline.
+    There should not be a need to instantiate this class directly; please use
+    :meth:`Client.create_registered_pipeline() <verta.Client.create_registered_piepline>`
+    for creating a new pipeline, or
+    :meth:`Client.get-registered_pipeline() <verta.Client.get_registered_piepline>`
+    for fetching existing pipelines.
+
+    Attributes
+    ----------
+        name: str
+            Name of this pipeline.
+        id: int
+            Auto-assigned ID of this Pipeline.
+        pipeline_graph: :class:`~verta.pipeline.PipelineGraph`
+            PipelineGraph object containing all possible steps in the Pipline.
+    """
+
+    def __init__(
+        self,
+        registered_model_version: RegisteredModelVersion,
+        pipeline_graph: PipelineGraph,
+    ):
+        """
+        Create a Pipeline instance from an existing RegisteredModelVersion object
+        and the provided pipeline graph. Name and ID are captured once upon creation
+        to avoid additional HTTP calls to refresh the cache of the RMV, because
+        pipelines are immutable.
+        """
+        self._registered_model_version = registered_model_version
+        self._name = self._registered_model_version.name
```

Comment:
Note that RMV names can be changed, so binding the name to a `RegisteredPipeline`'s state like this as if it were immutable runs the risk of inconsistency.

Reply:
Ok.  I'll tackle this with the change to the RMV name stuff above.  The refreshing of the cache for RMV causes a lot of headaches for mocking responses in the unit tests, but I think that can be solved with some tinkering.

Reply:
Given that we are accessing the `name` attribute of the RMV object when returning the `name` attribute of the pipeline, would the cache refresh process on the RMV pick up the name change?

Reply:
We'll be setting the name of the RMV for the pipeline via the client method, so the RMV name essentially will be the "pipeline name".  If a user changes the name of the RMV via the UI, that change should also be reflected in the Pipeline right?  The pipeline definition only contains an ID, so we wouldn't have to touch the artifact in that case, I don't think.

Reply:
Dropped the unnecessary variables in favor of just using the RMV so that we stay current with any name changes made in the UI.